### PR TITLE
Replace IClock with TimeProvider

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

Replaces the custom `IClock` with the `TimeProvider.System` 

## Commits

- refactor(time API)!: replace IClock and SystemClock with TimeProvider
- chore: update changelog for v3.2.0-preview.2 [skip ci]

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [x] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: test/UlidType.Tests/UlidTests.TypesAndData.cs [hunk 1]
```diff
@@ -57,13 +57,14 @@ class Test_IUlidRandomProvider : IUlidRandomProvider
...
     {
-        private readonly long _unixTimeMilliseconds;
+        private readonly DateTimeOffset _now;
 
-        public Test_IClock(long param) => _unixTimeMilliseconds = param;
+        public Test_TimeProvider(long param)
+            => _now = DateTimeOffset.FromUnixTimeMilliseconds(param);
 
-        public long UnixTimeMilliseconds() => _unixTimeMilliseconds;
+        public override DateTimeOffset GetUtcNow() => _now;
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply test/UlidType.Tests/UlidTests.TypesAndData.cs]: OK

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,14 @@
 # Changelog
 
+## v3.2.0-preview.2 - 2026-05-20
+
+### Internal
+
+- Bump the minor-and-patch group with 1 update
+- Update package dependencies and add Copilot instructions
+- Update Copilot instructions header to reflect package name change
+- rebuild *.lock.json
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply CHANGELOG.md:10]: Will be done by release pre-release
